cmd/deploy.go

```diff
@@ -103,26 +109,51 @@ generates a deployment manifest (yaml) file if one is not present, and uses it t
 			return errors.Errorf("Failed reading file %s", configFile)
 		}
 		defer yamlFile.Close()
+		var foundCreateKnativeTag, isKnativeService bool
+
 		scanner := bufio.NewScanner(yamlFile)
 		scanner.Split(bufio.ScanLines)
 		var txtlines []string
 		for scanner.Scan() {
+			var localFoundCreateKnativeTag bool
 			line := scanner.Text()
 			if strings.Contains(line, "applicationImage:") {
 				index := strings.Index(line, ": ")
 				start := line[0:(index + 2)]
 				line = start + deployImage
 				Info.log("Using applicationImage of: ", deployImage)
 			}
-			txtlines = append(txtlines, line)
```

Comment:
I would suggest the following:
```
for ... {
if strings.Contains(line, "createKnativeService") { //it's the infamous flag
 foundCreateKnativeTag = true
 if strings.Contains(line, "true") && !knative {
  //replace with false 
 } else if strings.Contains(line, "false") && knative {
  //replace with "true"
 }
} 
append()
} //end loop
//check if !found and append in that case

Reply:
I think that is good, one alternative might be to just check if the tag is there and then regen using the value of knative.  But the above is OK and probably simpler.

I think the bigger problem I just thought of is the append if createKnativeService is not there... Do we know that someone has not hand edited the file and added another k8s object defintiion?  I think we need to ask @neeraj-laad what the limits of editing app-deploy.yaml are.

Reply:
Changes have been made.  We will discuss the editing of app-deploy.yaml on the scrum and see if more work is needed.